agent/doc/swdesign/README.md

```diff
@@ -216,6 +216,133 @@ Needs:
+The Ankaios agent supports an insecure communication channel to the Ankaios server. Communicating over insecure channels is convenient during an evaluation or a development phase.
+
+Tags:
+- AgentManager
+
+Needs:
+- impl
+- stest
+
+#### Agent establishes insecure connection on provided insecure cli argument
+`swdd~agent-establishes-insecure-communication-based-on-provided-insecure-cli-argument~1`
+
+Status: approved
+
+When the user starts the Ankaios agent with the arguments `-k` or `--insecure` and the Ankaios agent initializes the communication middleware, then the Ankaios agent shall instruct the communication middleware to use an insecure communication channel.
+
+Tags:
+- AgentManager
+- CommunicationMiddleware
+
+Needs:
+- impl
+
+#### Agent provides file paths for mTLS to communication middleware
+`swdd~agent-provides-file-paths-to-communication-middleware~1`
+
+Status: approved
+
+When the file paths of the root certificate, the agent certificate and the agent key is provided upon startup, then the Ankaios agent shall provide each of these file paths to the communication middleware.
+
+Comment:
+The communication middleware parses the content of the files.
+
+Rationale:
+The communication middleware is responsible for establishing a secure communication channel.
+
+Tags:
+- AgentManager
+- CommunicationMiddleware
+
+Needs:
+- impl
+
+#### Agent fails upon missing pem file paths and insecure cli arguments
+`swdd~agent-fails-on-missing-file-paths-and-insecure-cli-arguments~1`
+
+Status: approved
+
+When the user starts the Ankaios agent without specifying the cli arguments for the mTLS PEM file paths and the insecure cli argument, then the Ankaios agent shall:
+* exit with a non zero exit code
+* output an error message containing the reason and a help message
+
+Rationale:
+The demand for an explicit communication mode prevents a user from unintentionally creating an Ankaios setup with a mixture of communication modes.
+
+Tags:
+- AgentManager
+
+Needs:
+- impl
+
+#### Agent supports environment variables
+`swdd~agent-supports-environment-variables~1`
+
+Status: approved
+
+The Ankaios agent shall support the usage of the following environment variables:
+
+- `ANKAGENT_CA_PEM`, for providing the file path to a certificate authority in PEM format,
+- `ANKAGENT_CRT_PEM`, for providing the file path to a certificate in PEM format,
+- `ANKAGENT_KEY_PEM`, for providing the file path to a key in PEM format,
+- `ANKAGENT_INSECURE`, for providing a boolean to enable/disable the insecure communication
+
+Rationale:
+This avoids passing cli arguments when starting an agent, which improves the usability.
+
+Tags:
+- AgentManager
+
+Needs:
+- impl
+- stest
+
+#### Agent prioritizes cli argument over environment variable
+`swdd~agent-prioritizes-cli-argument-over-environment-variable~1`
+
+Status: approved
+
+Command line arguments provided to the the Ankaios agent shall overwrite environment variables.
+
+Tags:
+- AgentManager
+
+Needs:
+- impl
+
```

Comment:
I would combine this too:
```suggestion
#### Agent supports environment variables
`swdd~agent-supports-environment-variables~1`

Status: approved

The Ankaios agent shall support the usage of the following environment variables which have lower priority compared to explicitly provided cli arguments:

- `ANKAGENT_CA_PEM`, for providing the file path to a certificate authority in PEM format,
- `ANKAGENT_CRT_PEM`, for providing the file path to a certificate in PEM format,
- `ANKAGENT_KEY_PEM`, for providing the file path to a key in PEM format,
- `ANKAGENT_INSECURE`, for providing a Boolean to enable/disable the insecure communication

Rationale:
Environment variables improve the usability, but are less important than explicitly provided configurations.

Tags:
- AgentManager

Needs:
- impl
- stest

```

Reply:
Ok I have made it the same way like in the CLI previously. I will change it in all components to a combined requirement.